packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -478,7 +498,7 @@ class SliverChildListDelegate extends SliverChildDelegate {
   /// The [children], [addAutomaticKeepAlives], [addRepaintBoundaries],
   /// [addSemanticIndexes], and [semanticIndexCallback] arguments must not be
   /// null.
-  const SliverChildListDelegate(
+  SliverChildListDelegate(
```

Comment:
I just notice this might be a breaking change. We cannot use const constructor if we want to keep state in this class. I think we have three choice here
1, keep this as breaking change
2, let user pass in the _findItemIndex, and just use it to look up index
3, we do not keep state here. _findItemIndex will just do a linear search on children and return index. We can, however, cache the result in SliverMultiBoxAdaptorElement

Reply:
As discussed with Ian, let's remove the const on this constructor and introduce a new const constructor with a new name that preserves the old behavior.

Reply:
done

Reply:
done